### PR TITLE
Initial dataset request implementation

### PR DESCRIPTION
**Pull request by emregoynugur**
I tried "mocking" the dataset requests from the client based on our discussions. Please let me know, if I am heading in the right direction :)

### CODE REVIEW

**Comment by bendecoste** [comment]
Don't merge yet -- ongoing discussion in slack #grid channel

**Comment by emregoynugur** [comment]
I think @bendecoste  is going to do a small change. He was planning to fix the method signature to use the values in the dictionary.

**Comment by iamtrask** [comment]
I really like how this is coming together - am I right in saying it's pretty close? It'd be good to merge it now - there will be lots of changes coming in the next day or so.

File: grid/adapter/adapter.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        parsed = np.frombuffer(data, dtype=np.uint8, offset=16)
+        return parsed.reshape(length, num_rows * num_cols)
+
+
+def get_int(b):
+    return int(codecs.encode(b, 'hex'), 16)
+
+
+def get_training_data(target):
+	if target == "mnist":
```
Comment (jvmancuso) [inline grid/adapter/adapter.py]: Target here is task name, correct?

File: grid/lib/utils.py [hunk 2]
```diff
@@ -17,15 +17,15 @@ def ipfs2keras(model_addr):
     return deserialize_keras_model(model_bin)
 
 def serialize_keras_model(model):
-    model.save('temp_model.h5')
-    with open('temp_model.h5','rb') as f:
+    model.save('mnist.h5')
```
Comment (jvmancuso) [inline grid/lib/utils.py]: I don't think this should have changed, it's supposed to be for general models.

File: grid/adapter/adapter.py (hunk 1 above)
Comment (jvmancuso) [reply grid/adapter/adapter.py]: I'm a little concerned that this isn't robust enough.  We can't be adding tasks to these functions for every dataset/task out there.

**Pull request closed without merging**